File: Rentify.Services/Database/Expense.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Rentify.Services.Database
{
    public class Expense
    {
        [Key]
        public int Id { get; set; }

        [ForeignKey(nameof(UserId))]
        public int UserId { get; set; }
        public User User { get; set; }

        public int? PropertyId { get; set; }
        public Property? Property { get; set; }

        public string Description { get; set; }
        public decimal Amount { get; set; }
        public DateTime Date { get; set; }
        public string Category { get; set; }
        public DateTime CreatedAt { get; set; }
    }
}

File: UI/rentify_desktop/lib/models/expense.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:rentify_desktop/models/property.dart';
import 'package:rentify_desktop/models/user.dart';

part 'expense.g.dart';

@JsonSerializable(explicitToJson: true)
class Expense {
  final int id;
  final int userId;
  final User? user;
  final int? propertyId;
  final Property? property;
  final String description;
  final double amount;
  final DateTime date;
  final String category;
  final DateTime createdAt;

  Expense({
    required this.id,
    required this.userId,
    this.user,
    this.propertyId,
    this.property,
    required this.description,
    required this.amount,
    required this.date,
    required this.category,
    required this.createdAt,
  });

  factory Expense.fromJson(Map<String, dynamic> json) =>
      _$ExpenseFromJson(json);

  Map<String, dynamic> toJson() => _$ExpenseToJson(this);
}
